src/controller/PingClient.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines an object for a CHIP Ping client which using Echo
+ *      Protocol to measure packet loss across network paths for a paired device.
+ */
+
+#pragma once
+
+#include <messaging/ExchangeMgr.h>
+#include <protocols/echo/Echo.h>
+
+namespace chip {
+namespace Controller {
+
+class PingClient : public Protocols::Echo::EchoDelegate
```

Comment:
Does this handle both IPv4 and IPv6 ping? Could you add a comment to clarify where this is going out? Is there an address preference?

Reply:
This ping is on the application level, not on the ip level. It is depending on how to connect the peer device before you issue the ping command.

Reply:
If this isn't actually a ping, could it be called something else? Ping is pretty specific.